### PR TITLE
Build: Make NonStop shared libraries only export selected symbols 

### PR DESCRIPTION
**Pull request by levitte**

### CODE REVIEW

**Comment by levitte** [comment]
This currently contains the commits from #12858. When that's merged, this will be rebased and only contain the intended commit.

@rsbecker, it would be nice if you could test and confirm that it works as intended

**Comment by rsbeckerca** [comment]
> 
> 
> This currently contains the commits from #12858. When that's merged, this will be rebased and only contain the intended commit.
> 
> @rsbecker, it would be nice if you could test and confirm that it works as intended

Testing this as part of levitte/NonStop-dont-export-all, correct?

**Comment by levitte** [comment]
> Testing this as part of levitte/NonStop-dont-export-all, correct?

That's the branch of this PR, yes

**Comment by levitte** [comment]
Hmmm, for testing, this will probably hit the command line length that #12960 fixes (because this branch is parallel to that one).  Maybe this branch should be rebased on top of the #12960 branch instead, so it can be tested fully...

**Comment by rsbeckerca** [comment]
> 
> 
> Hmmm, for testing, this will probably hit the command line length that #12960 fixes (because this branch is parallel to that one). Maybe this branch should be rebased on top of the #12960 branch instead, so it can be tested fully...

Yes, it did hit it. I can manually patch the .so builds, so that should not be too bad. The awkwardness is the time it takes to do the build/test. I am deliberately using our older-slower system to ensure that it works there. Currently testing this particular branch.

**Comment by rsbeckerca** [comment]
I am into linking objects and that seems to be working fine. No undefined externals.
```
openssl-3.0/apps: ./openssl version
OpenSSL 3.0.0-alpha7-dev  (Library: OpenSSL 3.0.0-alpha7-dev )
```

**Comment by rsbeckerca** [comment]
Let's get this thing into the main branch and go on with the next task.

**Comment by levitte** [comment]
@openssl/committers, it seems it's time for a final review

File: Configurations/50-nonstop.conf
Comment (paulidale) [inline Configurations/50-nonstop.conf:174]: `s/entried/entries/`

**Review by paulidale — APPROVED** [review]
A comment typo.

Plus [WIP] is still in the title.

**Comment by levitte** [comment]
> Plus [WIP] is still in the title.

Yes, until #12933 gets merged...  which happened a minute ago, so I'm rebasing this

**Comment by levitte** [comment]
Squashed and rebased, and includes the fix of the nit @paulidale picked on

**Comment by levitte** [comment]
@paulidale, re-confirm please, and don't forget to label :wink:

**Review by paulidale — APPROVED** [review]

**Comment by levitte** [comment]
Merged

7a032be7f293bd80e3fe18c5568cf382b0b79543 Build: Make NonStop shared libraries only export selected symbols

**Pull request closed without merging**
